client/my-sites/stats/stats-post-detail/index.jsx

```diff
@@ -37,16 +38,14 @@ export default React.createClass( {
 	render() {
 		let title;
 
-		if ( this.props.postViewsList.response.post && this.props.postViewsList.response.post.post_title ) {
-			title = this.translate( 'Stats for %(posttitle)s', {
-				comment: 'Title of the individual post stats page.',
-				args: {
-					posttitle: this.props.postViewsList.response.post.post_title
-				}
-			} );
-		}
+		const post = this.props.postViewsList.response.post;
+		const postOnRecord = post && post.post_title !== null;
 
-		if ( this.props.postViewsList.isError() ) {
+		if ( postOnRecord ) {
+			if ( typeof post.post_title === 'string' && post.post_title.length ) {
```

Comment:
@timmyc -- that was intentional to cover the case of an empty post title. Doing something like this on L#47 would make it more clear, I think:

```
+                       } else {
+                               title = this.translate( '(no title)' );
```
...instead of just rendering an empty title component.  You agree?